src/aind_hcr_data_transformation/compress/czi_to_zarr_purezarr.py

```diff
@@ -0,0 +1,458 @@
+    chunk_size: List[int],
+    scale_factor: List[int],
+    n_lvls: int,
+    channel_name: str,
+    stack_name: str,
+    compressor_kwargs: dict,
+    bucket_name: Optional[str] = None,
+    downsample_mode: Optional[str] = "mean",  # Only mean supported
+    overwrite_existing_data: bool = False,
+    macro_block_target_mb: int = 12800,
+    max_workers: int = 4,  # New parameter for parallel writes
+):
+    """
+    Write CZI stack to OME-Zarr format using pure zarr v3 (simplified implementation).
+    
+    This version is simplified for maintainability and follows SmartSPIM patterns:
+    - Always uses S3 storage
+    - Always uses Blosc compression  
+    - Always builds pyramid with xarray_multiscale
+    - Optimized parallel writing strategy
+    - Memory-efficient streaming pyramid building
+    - Parallel tile uploads within each node
+    
+    Args:
+        max_workers: Maximum number of threads for parallel chunk uploads (default: 12)
+        Other parameters: Same as TensorStore backend for API compatibility
+    
+    Performance optimizations:
+    - Uses da.from_zarr() instead of loading full arrays into memory
+    - Parallel pyramid level computation and writing  
+    - Batched parallel writes to reduce thread overhead
+    - Optimized dask computation with persist() calls
+    """
+    
+    # Handle S3 path normalization
+    if output_path.startswith("s3://"):
+        bucket, key_base = _split_s3_path(output_path)
+        if bucket_name and bucket_name != bucket:
+            logging.warning(f"bucket_name {bucket_name} differs from path bucket {bucket}. Using {bucket}")
+        bucket_name = bucket
+        output_path = key_base
+    
+    if bucket_name is None:
+        raise ValueError("bucket_name must be provided or embedded in s3:// path")
+    
+    # Construct dataset path
+    dataset_path = f"{output_path.strip('/')}/{stack_name}"
+    
+    # Check for existing dataset
+    slurm_id = os.environ.get("SLURM_ARRAY_TASK_ID") or os.environ.get("SLURM_JOB_ID")
+    if _dataset_exists(bucket_name, dataset_path) and not overwrite_existing_data:
+        logging.info(f"[purezarr]{' [SLURM '+slurm_id+']' if slurm_id else ''} "
+                    f"Skipping existing dataset at s3://{bucket_name}/{dataset_path}")
+        return
+    
+    start_time = time.time()
+    
+    # Log ignored parameters for clarity
+    if shard_size:
+        logging.debug("shard_size parameter ignored (using chunk_size)")
+    if downsample_mode and downsample_mode != "mean":
+        logging.warning(f"Only mean downsampling supported, ignoring {downsample_mode}")
+    
+    with czifile.CziFile(str(czi_path)) as czi:
+        # Get 5D shape (T,C,Z,Y,X) - pad with 1s as needed  
+        # Access shape through asarray to avoid lazyattr issues
+        sample_data = czi.asarray()[:1]  # Get just first element to get shape
+        czi_shape = sample_data.shape[:-len(sample_data.shape)+len(czi.asarray().shape)]
```

Comment:
can you squeeze and get the shape? CZI files have these very long shapes, might be easier